### PR TITLE
Improve the test reliability by modularization

### PR DESCRIPTION
**Pull request by weiiwang01**
A variety of measures to improve the reliability of integration tests include:

- Increasing the margin of safety when waiting
- Providing a more accurate timeout deadline in the charm code
- Reducing the amount of unnecessary and insignificant integration tests
- Modularizing integration tests

### CODE REVIEW

**Review by arturo-seijas — DISMISSED** [review]
Nice! May I also suggest to make the changes specified in https://github.com/canonical/operator-workflows/pull/162?

**Comment by weiiwang01** [comment]
> Nice! May I also suggest to make the changes specified in [canonical/operator-workflows#162](https://github.com/canonical/operator-workflows/pull/162)?
 
I have already added the `--charm-file` option, but the wordpress is using the `integration_test_run` workflow instead of `integration_test` workflow for extra customization. I think I remove the `db_*` configuration from the charm which will allow us to switch to the `integration_test` workflow in the following pull request.

**Comment by arturo-seijas** [comment]
> > Nice! May I also suggest to make the changes specified in [canonical/operator-workflows#162](https://github.com/canonical/operator-workflows/pull/162)?
> 
> I have already added the `--charm-file` option, but the wordpress is using the `integration_test_run` workflow instead of `integration_test` workflow for extra customization. I think I remove the `db_*` configuration from the charm which will allow us to switch to the `integration_test` workflow in the following pull request.

I hadn't noticed that. That sounds like a good idea

**Review by amandahla — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

File: tests/integration/test_error.py [hunk 1]
```diff
@@ -4,28 +4,12 @@
...
-            "db_name": "test_db_name",
-            "db_user": "test_db_user",
-            "db_password": "test_db_password",
-        }
-    ],
-    indirect=True,
-    scope="function",
-)
-async def test_incorrect_db_config(ops_test: pytest_operator.plugin.OpsTest, application_name):
+@pytest.mark.usefixtures("wordpress")
```
Comment (jdkandersson) [inline tests/integration/test_error.py:11]: Is the `usefixtures` required given that it is being taken as an argument?

**Review by jdkandersson — COMMENTED** [review]
Would be good for @yanksyoon so also review this PR

File: tests/integration/test_error.py (hunk 1 above)
Comment (weiiwang01) [reply tests/integration/test_error.py]: Yes, it's unnecessary, removed, thanks!

**Review by gtrkiller — APPROVED** [review]
LGTM

**Review by arturo-seijas — APPROVED** [review]
Great!

**Pull request merged**